### PR TITLE
AKCORE-9: Add support for describe share groups in bin script.

### PR DESCRIPTION
**Pull request by smjn**
```
➜  ccs-kafka git:(AKCORE-9-cli) ✗ bin/kafka-share-groups.sh --bootstrap-server localhost:9092 --describe --group group-share-x                                                     [2:05:13]

GROUP_ID             COORDINATOR_NODE     OFFSETS
group-share-x        1                    topic-share-y-0=>878
group-share-x        1                    topic-share-x-0=>665
➜  ccs-kafka git:(AKCORE-9-cli) ✗ bin/kafka-share-groups.sh --bootstrap-server localhost:9092 --describe --group group-share                                                       [2:05:16]

GROUP_ID             COORDINATOR_NODE     OFFSETS
group-share          1
➜  ccs-kafka git:(AKCORE-9-cli) ✗ bin/kafka-share-groups.sh --bootstrap-server localhost:9092 --describe --group group-share-x --offsets                                           [2:05:19]

GROUP_ID             COORDINATOR_NODE     OFFSETS
group-share-x        1                    topic-share-y-0=>883
group-share-x        1                    topic-share-x-0=>670
➜  ccs-kafka git:(AKCORE-9-cli) ✗ bin/kafka-share-groups.sh --bootstrap-server localhost:9092 --describe --group group-share-x --state                                             [2:05:27]

GROUP_ID             COORDINATOR_NODE     OFFSETS              STATE
group-share-x        1                    topic-share-y-0=>886 Stable
group-share-x        1                    topic-share-x-0=>673 Stable
➜  ccs-kafka git:(AKCORE-9-cli) ✗ bin/kafka-share-groups.sh --bootstrap-server localhost:9092 --describe --group group-share-x --members                                           [2:05:32]

MEMBER_ID              CLIENT_ID              HOST                   ASSIGNMENT
eYP6tgjWT3amoKW19QKaUw console-share-consumer /127.0.0.1             topic-share-x:0
rOptwrnfQoibv62vHzfiyw console-share-consumer /127.0.0.1             topic-share-y:0
```

Screenshot 2024-02-16 at 2 06 43 AM

### CODE REVIEW

File: tools/src/main/java/org/apache/kafka/tools/ShareGroupsCommand.java [hunk 1]
```diff
@@ -68,6 +71,9 @@ public static void run(ShareGroupCommandOptions opts) {
             // Currently the tool only supports listing of share groups
```
Comment (apoorvmittal10) [inline tools/src/main/java/org/apache/kafka/tools/ShareGroupsCommand.java:71]: Nit: might want to correct this as well.

**Review by apoorvmittal10 — APPROVED** [review]

**Comment by apoorvmittal10** [comment]
Would have been good if there would be a response from the terminal in description which shows the output. It would be then easy to see the print statements and details.

File: tools/src/main/java/org/apache/kafka/tools/ShareGroupsCommand.java [hunk 2]
```diff
@@ -68,6 +71,9 @@ public static void run(ShareGroupCommandOptions opts) {
...
             if (opts.options.has(opts.listOpt))
```
Comment (AndrewJSchofield) [inline tools/src/main/java/org/apache/kafka/tools/ShareGroupsCommand.java:72]: Consistent formatting for the entire if-else-if please.

File: tools/src/main/java/org/apache/kafka/tools/ShareGroupsCommand.java [hunk 3]
```diff
@@ -155,6 +161,94 @@ private void printGroupStates(List<Tuple2<String, String>> groupsAndStates) {
...
+            System.out.println();
+        }
+
+        private void printMemberDetails(Collection<MemberDescription> members) {
+            List<List<String>> lineItems = new ArrayList<>();
+            int maxLen = 20;
+            for (MemberDescription member : members) {
+                List<String> lineItem = new ArrayList<>();
+                lineItem.add(member.consumerId());
+                lineItem.add(member.groupInstanceId().isPresent() ? member.groupInstanceId().get() : "");
```
Comment (AndrewJSchofield) [inline tools/src/main/java/org/apache/kafka/tools/ShareGroupsCommand.java:225]: There is no group instance ID for share groups, so this item can be omitted from the output entirely.

**Review by AndrewJSchofield — CHANGES_REQUESTED** [review]
Overall, lgtm, but I did not manage to get it to display a share group. There's something wrong in the stack somewhere I think.

File: tools/src/main/java/org/apache/kafka/tools/ShareGroupsCommand.java [hunk 4]
```diff
@@ -155,6 +161,94 @@ private void printGroupStates(List<Tuple2<String, String>> groupsAndStates) {
...
+                lineItem.add(member.clientId());
+                lineItem.add(member.host());
+                lineItem.add(member.assignment().topicPartitions().stream().map(part -> part.topic() + ":" + part.partition()).collect(Collectors.joining(",")));
+                lineItem.add(member.targetAssignment().isPresent()
```
Comment (AndrewJSchofield) [inline tools/src/main/java/org/apache/kafka/tools/ShareGroupsCommand.java:229]: Target assignment also does not apply to share groups and can be omitted.

**Comment by AndrewJSchofield** [comment]
@smjn I needed to make a change to `DescribeShareGroupsHandler`. Once that was made, this code works fine and I can use `kafka-share-groups.sh` to display a share group.

**Comment by smjn** [comment]
> Overall, lgtm, but I did not manage to get it to display a share group. There's something wrong in the stack somewhere I think.

Found the issue - Will raise commit

File: tools/src/main/java/org/apache/kafka/tools/ShareGroupsCommand.java (hunk 2 above)
Comment (smjn) [reply tools/src/main/java/org/apache/kafka/tools/ShareGroupsCommand.java:72]: will fix, guess it was part of previous PR

**Review by AndrewJSchofield — APPROVED** [review]
Thanks for the PR. Looks good to me.

**Pull request merged**